ddl/ddl_worker.go

```diff
@@ -520,6 +520,11 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx) error {
 		d.mu.hook.OnJobUpdated(job)
 		d.mu.RUnlock()
 
+		// d.mu.hook is initialed from domain, which will force the owner host update schema diff synchronously.
+		d.mu.RLock()
+		d.mu.hook.OnSchemaStateChanged()
```

Comment:
1. OnSchemaStateChanged should be before OnJobUpdated
2. Call it only in test, otherwise it would affect performance and produce too many logs.

Reply:
addressed